compiler/src/iree/compiler/Codegen/LLVMGPU/TransformExtensions/LLVMGPUExtensions.cpp

```diff
@@ -112,15 +113,10 @@ FailureOr<SmallVector<OpFoldResult>> rewriteForeachThreadToGpu(
         "scf.foreach_thread with rank > 3 does not lower to gpu.thread");
 
   auto maybeWorkgroupSizes = getNumThreads(rewriter, foreachThreadOp);
-  if (failed(maybeWorkgroupSizes) ||
-      llvm::any_of(*maybeWorkgroupSizes, [](OpFoldResult ofr) {
-        return !getConstantIntValue(ofr).has_value();
-      }))
-    return foreachThreadOp->emitError("unsupported dynamic workgroup size");
+  if (failed(maybeWorkgroupSizes))
+    return foreachThreadOp->emitError("can't extract number of threads");
 
-  SmallVector<int64_t> workgroupSizes;
-  for (OpFoldResult ofr : *maybeWorkgroupSizes)
-    workgroupSizes.push_back(getConstantIntValue(ofr).value());
```

Comment:
Do we have a test to cover the dynamic case? Otherwise, looks great.

Reply:
This is a good, I thought I needed this but I realize now with my latest changes I always have a constant workgroup size. I reverted that for now until it gets properly tested.